qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/inlayhint/InlayHintASTVistor.java

```diff
@@ -239,8 +250,31 @@ private static CompletableFuture<ResolvedJavaTypeInfo> getJavaType(Parameter par
 		return javaType;
 	}
 
-	private static void updateJavaType(InlayHint hint, ResolvedJavaTypeInfo javaType) {
-		String type = javaType != null ? javaType.getJavaElementSimpleType() : "?";
-		hint.setLabel(Either.forLeft(":" + type));
+	private void updateJavaType(InlayHint hint, ResolvedJavaTypeInfo javaType, String projectUri) {
+		String type = getLabel(javaType);
+		if (javaType != null && canSupportJavaDefinition) {
+			// first part : ":"
+			InlayHintLabelPart separatorPart = new InlayHintLabelPart(":");
+			// second part label : clickable Java type (ex : String)
+			InlayHintLabelPart javaTypePart = new InlayHintLabelPart(type.toString());

```

Comment:
`type` is already a string (the label string for `javaType`), so there is not need to call toString()

Reply:
Good catch, fixed